### PR TITLE
1606837: Add spellchecking to docs

### PR DESCRIPTION
**Pull request by mdboom**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ To contact us you can:
 - Send an email to *glean-team@mozilla.com*.
 - The Glean SDK team is: *:janerik*, *:dexter*, *:travis*, *:mdroettboom*, *:gfritzsche*, *:chutten*, *:brizental*.
 
-The source code is available [on Github](https://github.com/mozilla/glean/).
+The source code is available [on GitHub](https://github.com/mozilla/glean/).
```
Comment (badboy) [inline docs/README.md]: :D yes!

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -64,7 +65,7 @@
         - [Payload format](dev/core/internal/payload.md).
         - [Directory structure](dev/core/internal/directory-structure.md)
         - [Debug Pings](dev/core/internal/debug-pings.md)
-    - [Howtos](dev/howtos/index.md)
+    - [HOWTOs](dev/howtos/index.md)
```
Comment (badboy) [inline docs/SUMMARY.md]: Not a big fan of all-uppercasing this (but also not a blocker)

File: docs/dev/android/setup-android-build-environment.md [hunk 3]
```diff
@@ -4,12 +4,12 @@
 
 This document describes how to make local builds of the Android bindings in
 this repository. Most consumers of these bindings *do not* need to follow
-this process, but will instead use pre-built bindings [todo: link to this]
+this process, but will instead use pre-built bindings [TODO: link to this]
```
Comment (badboy) [inline docs/dev/android/setup-android-build-environment.md]: huh? we should ... add the link (or file a bug)

File: docs/dev/android/setup-android-build-environment.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ If you want to install the NDK manually:
 1. Download NDK r20 from <https://developer.android.com/ndk/downloads>.
 2. Extract it and put it somewhere (`$HOME/.android-ndk-r20` is a reasonable choice, but it doesn't matter).
 3. Set `ANDROID_NDK_ROOT` to this path.
-    * Set `ANDROID_NDK_HOME` to match `ANDROID_NDK_ROOT`, for compatibility with some android grandle plugins.
```
Comment (badboy) [inline docs/dev/android/setup-android-build-environment.md]: grandle hell.

Nope, doesn't have the same flow.

File: docs/dev/ffi/when-to-use-what-in-the-ffi.md [hunk 5]
```diff
@@ -15,8 +15,8 @@ there are two kinds of values you may want to pass over the FFI.
 
 ## Types with identity
 
-Examples of this are things like database connections, the FirefoxAccounts
-struct, etc. These types are complex, implemented in rust, and it's not
+Examples of this are things like database connections, the `FirefoxAccounts`
+struct, etc. These types are complex, implemented in Rust, and it's not
```
Comment (badboy) [inline docs/dev/ffi/when-to-use-what-in-the-ffi.md]: Hah, we might want to change this to name our own experience.

File: docs/SUMMARY.md (hunk 2 above)
Comment (mdboom) [reply docs/SUMMARY.md]: Yeah -- I'll revert this one (and add it to the dictionary).

File: docs/dev/android/setup-android-build-environment.md (hunk 3 above)
Comment (mdboom) [reply docs/dev/android/setup-android-build-environment.md]: I guess that should go here? https://github.com/mozilla/glean/releases  Or link to the wrapper in a-c?

File: docs/dev/ffi/when-to-use-what-in-the-ffi.md (hunk 5 above)
Comment (mdboom) [reply docs/dev/ffi/when-to-use-what-in-the-ffi.md]: Yeah -- but I guess as a follow-on bug...

File: docs/dev/android/setup-android-build-environment.md (hunk 3 above)
Comment (Dexterp37) [reply docs/dev/android/setup-android-build-environment.md]: Or maybe just link to [our other docs](https://mozilla.github.io/glean/book/user/adding-glean-to-your-project.html) ?

File: docs/dev/android/setup-android-build-environment.md [hunk 6]
```diff
@@ -4,12 +4,12 @@
...
 This document describes how to make local builds of the Android bindings in
 this repository. Most consumers of these bindings *do not* need to follow
-this process, but will instead use pre-built bindings [todo: link to this]
+this process, but will instead use pre-built bindings [TODO: link to this]
 
 ## Prepare your build environment
 
 Typically, this process only needs to be run once, although periodically you
-may need to repeat some steps (eg, rust updates should be done periodically)
+may need to repeat some steps (eg., rust updates should be done periodically)
```
Comment (Dexterp37) [inline docs/dev/android/setup-android-build-environment.md]: Shouldn't this be `e.g.`?

File: docs/dev/docs.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
+# Developing documentation
```
Comment (Dexterp37) [inline docs/dev/docs.md]: Do we want to mention, somewhere in this document, that we also have some other off-site docs for the "Glean as a product" thing? I mean [these docs](https://docs.telemetry.mozilla.org/concepts/glean/glean.html).

File: docs/dev/android/setup-android-build-environment.md (hunk 6 above)
Comment (mdboom) [reply docs/dev/android/setup-android-build-environment.md]: Indeed.  Good catch.

File: docs/dev/docs.md (hunk 7 above)
Comment (mdboom) [reply docs/dev/docs.md]: Good point.

**Pull request merged**